src/models.py

```diff
@@ -0,0 +1,100 @@
+from datetime import datetime
+from enum import Enum
+from pydantic import (
+    BaseModel,
+    PositiveInt,
+    StringConstraints,
+    EmailStr,
+    field_validator,
+    model_validator,
+)
+from typing_extensions import Annotated, Optional, Self
+from tools.tools import short_uuid4_generator
+from pydantic_extra_types.country import CountryAlpha2
+
+
+class ActivityTypes(str, Enum):
+    click = "click"
+    login = "login"
+    logout = "logout"
+    purchase = "purchase"
+
+
+class SuperUserRoles(str, Enum):
+    admin = "admin"
+    moderator = "moderator"
+    support = "support"
+
+
+class User(BaseModel, validate_assignment=True):
+    user_id: PositiveInt
+    username: Annotated[str, StringConstraints(min_length=2, pattern=r"^[a-zA-Z]*$")]
+    email: EmailStr
+    age: Optional[PositiveInt]
+    country: Optional[CountryAlpha2]
+
+    @model_validator(mode="after")
+    def check_if_both_age_and_country(self) -> Self:
+        age = self.age
+        country = self.country
+        if age is None and country is None:
+            raise ValueError("Enter at least age or country")
+        return self
+
+
+class SuperUser(User, validate_assignment=True):
+    superuser_id: PositiveInt
+    role: SuperUserRoles
+
+
+class Activity(BaseModel, validate_assignment=True):
+    activity_id: PositiveInt
+    time: datetime
+    user_id: PositiveInt
+    activity_type: ActivityTypes
+    activity_details: str
+
+    @field_validator("time", mode="before")
+    def prevalidate_datetime(cls, entry):
+        if isinstance(entry, int):
+            raise ValueError("Integers are not valid datetime entries")
+        if isinstance(entry, str):
+            try:
+                datetime.strptime(entry, "%Y-%m-%d %H:%M:%S")
+            except ValueError:
+                raise ValueError(
+                    'Incorrect data format, should be "YYYY-MM-DD HH:MM:SS" (UTC time)'
+                )
+        return entry
+
+
+def create_user(
+    username, email, age=None, country=None
+) -> User:  # TODO: should this be the class' __init__ method?
```

Comment:
no need for `__init__` at the moment, these models are very straightforward to construct :)
I'd argue that you don't even need these extra functions to create a user or an activity, but if you think these are helpful functions for something else, then you can leave it.